### PR TITLE
Adjust UI and complete CFD history position modal

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP

- [x] CandlestickChart
- [x] 001	NavBar
- [x] 117	NavBarMobile
- [x] GlobalContext
- [x] 032	CryptoCategory
- [x] 033	CryptoCard
- [x] 037	TickerSelectorBox
- [x] 041	CryptoNewsSection
- [x] 042	CryptoNewsItem
- [x] CryptoSummary 
- [x] HeroDescription
- [x] HistoryPositionItem
- [x] HistoryPositionModal
- [x] HistorySubTab 
- [x] HistorySubTabMobile 
- [x] MarketSection 
- [x] OpenPositionItem
- [x] OpenSubTab 
- [x] OpenSubTabMobile 
- [x] OrderSection
- [x] PositionClosedModal
- [x] CryptoCard 
- [x] CryptoNewsItem 
- [x] CryptoNewsSection 
- [x] PositionDetailsModal
- [x] PositionOpenModal

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 0
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/constants/display.ts [hunk 1]
```diff
@@ -116,7 +116,7 @@ export const CRYPTO_CARD_COLORS = [
 export const TIDEBIT_BUTTON_CONFIG =
   'rounded px-5 py-2 text-base transition-all hover:opacity-90 text-white bg-tidebitTheme';
 
-export interface IDEV_TOAST_CONFIG {
+export interface IDevToastConfig {
   position: string;
```
Comment (arealclimber) [inline src/constants/display.ts:120]: 命名規則

File: src/components/history_position_modal/history_position_modal.tsx [hunk 2]
```diff
@@ -180,7 +180,7 @@ const HistoryPositionModal = ({
```
Comment (arealclimber) [inline src/components/history_position_modal/history_position_modal.tsx:180]: - 拿 `state`

File: src/contexts/global_context.tsx [hunk 3]
```diff
@@ -253,6 +276,21 @@ export const GlobalProvider = ({children}: IGlobalProvider) => {
   const [dataSignatureProcessModal, setDataSignatureProcessModal] =
     useState<IDataSignatrueProcessModal | null>();
 
+  const [visibleHistoryPositionModal, setVisibleHistoryPositionModal] = useState(false);
+  const [dataHistoryPositionModal, setDataHistoryPositionModal] =
+    useState<IDataHistoryPositionModal>({closedCfdDetails: dummyCloseCFDDetails});
```
Comment (arealclimber) [inline src/contexts/global_context.tsx:281]: - 拆開

File: src/components/history_position_item/history_position_item.tsx [hunk 4]
```diff
@@ -1,57 +1,68 @@
...
-        <div className="flex justify-between">
-          <div>
-            <div className="text-lightGray">11 Nov</div>
-            <div className="text-lightGray">15:05</div>
+      <div className="mt-3 text-xs hover:cursor-pointer" onClick={itemClickHandler}>
+        <div className="flex justify-center">
+          <div className="w-48px">
+            <div className="text-lightGray">
+              {timestampToString(closedCfdDetails.closedTimestamp).day}{' '}
+              {timestampToString(closedCfdDetails.closedTimestamp).abbreviatedMonth}{' '}
```
Comment (arealclimber) [inline src/components/history_position_item/history_position_item.tsx:46]:
```
const sth = timestampToString(closedCfdDetails.closedTimestamp)
sth.day
stg.abbreviatedMonth
```

File: src/components/history_position_modal/history_position_modal.tsx [hunk 5]
```diff
@@ -121,42 +120,43 @@ const HistoryPositionModal = ({
...
           </div>
 
           <div className={`${layoutInsideBorder}`}>
             <div className="text-lightGray">Open Time</div>
             <div className="">
-              {timestampToString(openCfdDetails?.openTimestamp ?? 0)[0]}{' '}
-              {timestampToString(openCfdDetails?.openTimestamp ?? 0)[1]}
+              {timestampToString(closedCfdDetails?.openTimestamp ?? 0).date}{' '}
+              {timestampToString(closedCfdDetails?.openTimestamp ?? 0).time}
             </div>
```
Comment (arealclimber) [inline src/components/history_position_modal/history_position_modal.tsx:139]: - 修改

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**